charmcraft/commands/store/__init__.py

```diff
@@ -322,9 +325,31 @@ def fill_parser(self, parser):
             '--release', action='append',
             help="The channel(s) to release to (this option can be indicated multiple times)")
 
+    def _validate_template_is_handled(self, filepath):
+        """Verify the zip does not have any file with the 'init' template TODO marker.
+
+        This is important to avoid uploading low-quality charms that are just
+        bootstrapped and not corrected.
+        """
+        # we're already sure we can open it ok
+        zf = zipfile.ZipFile(str(filepath))
+
+        tainted_filenames = []
+        for name in zf.namelist():
+            content = zf.read(name)
+            if INIT_TEMPLATE_TOKEN in content:
```

Comment:
Is it safe to assume that all files in the zipfile will be small enough to read to memory? Can we filter the possible candidates to have `TEMPLATE-TODO` by filename, or could it really be any file?

Reply:
Charm files tend to be very small, yes. We *could* filter by files (e.g., *.py, *.yaml, *.md, etc), but I wanted to keep this as decoupled as possible from the other part of the code.